pkg/certificate/providers/certmanager/certificate_manager.go

```diff
@@ -52,7 +52,7 @@ func (cm *CertManager) GetCertificate(cn certificate.CommonName) (certificate.Ce
 	if cert := cm.getFromCache(cn); cert != nil {
 		return cert, nil
 	}
-	return nil, fmt.Errorf("failed to find certificate with CN=%s", cn)
+	return nil, errCertNotFound
```

Comment:
This error is now being used in the vault and cert-manager packages. We could move the `errCertNotFound` definition to pkg/certificate/providers/errors.go

Reply:
Not sure for vault's case, but it creates an import cycle for cert-manager if we do this